File: product/src/main/scala/de/leanovate/dose/product/consul/ServiceNode.scala
package de.leanovate.dose.product.consul

import spray.json.{JsonFormat, DefaultJsonProtocol}

case class ServiceNode(
  Node: String,
  Address: String,
  ServiceID: String,
  ServiceName: String,
  ServiceTags: Seq[String],
  ServicePort: Int
  )

object ServiceNode extends DefaultJsonProtocol {
  implicit val serviceNodeFormat : JsonFormat[ServiceNode]= jsonFormat6(ServiceNode.apply)
}

File: customer/src/main/java/de/leanovate/dose/customer/consul/ServiceNode.java
package de.leanovate.dose.customer.consul;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;

import java.util.List;

@JsonIgnoreProperties(ignoreUnknown = true)
public class ServiceNode {
    public String Node;

    public String Address;

    public String ServiceID;

    public String ServiceName;

    public List<String> ServiceTags;

    public int ServicePort;

    @Override
    public String toString() {

        return "ServiceNode{" +
                "Node='" + Node + '\'' +
                ", Address='" + Address + '\'' +
                ", ServiceID='" + ServiceID + '\'' +
                ", ServiceName='" + ServiceName + '\'' +
                ", ServiceTags=" + ServiceTags +
                ", ServicePort=" + ServicePort +
                '}';
    }
}
